File: Support/Templates/cs/Files/ReadOnlyBindingList.cs
using System;
using System.Collections.Generic;
using Csla;

namespace Templates
{
  [Serializable]
  public class ReadOnlyBindingList :
    ReadOnlyBindingListBase<ReadOnlyBindingList, ReadOnlyChild>
  {
    private static void AddObjectAuthorizationRules()
    {
      // TODO: add authorization rules
      //AuthorizationRules.AllowGet(typeof(ReadOnlyBindingList), "Role");
    }

    [Fetch]
    private void Fetch(string criteria)
    {
      RaiseListChangedEvents = false;
      IsReadOnly = false;
      // TODO: load values
      object objectData = null;
      foreach (var child in (List<object>)objectData)
        Add(DataPortal.Fetch<ReadOnlyChild>(child));
      IsReadOnly = true;
      RaiseListChangedEvents = true;
    }
  }
}

File: Support/Templates/vb/Files/ReadOnlyBindingList.vb
Imports System
Imports System.Collections.Generic
Imports Csla
Imports Csla.Security

<Serializable()> _
Public Class ReadOnlyBindingList
  Inherits ReadOnlyBindingListBase(Of ReadOnlyBindingList, ReadOnlyChild)

  Private Shared Sub AddObjectAuthorizationRules()
    ' TODO: add authorization rules 
    ' AuthorizationRules.AllowGet(GetType(ReadOnlyList), "Role")
  End Sub

  <Fetch>
  Private Sub Fetch(ByVal criteria As String)
    RaiseListChangedEvents = False
    IsReadOnly = False
    ' TODO: load values 
    Dim objectData As Object = Nothing
    For Each child As Object In DirectCast(objectData, List(Of Object))
      Add(DataPortal.FetchChild(Of ReadOnlyChild)(child))
    Next
    IsReadOnly = True
    RaiseListChangedEvents = True
  End Sub

End Class
